lib/Myriad/Exception.pm

```diff
@@ -20,22 +17,17 @@ This is a rôle used for all exceptions throughout the framework.
 
 =cut
 
-no indirect qw(fatal);
-
-use Scalar::Util;
-use Role::Tiny;
-
-requires qw(category message);
+requires category;
+requires message;
 
 =head2 throw
 
 Instantiates a new exception and throws it (by calling L<perlfunc/die>).
 
 =cut
 
-sub throw {
-    my $self = shift;
-    $self = $self->new(@_) unless Scalar::Util::blessed($self);
+sub throw ($class, @args) {
```

Comment:
shouldn't be defined as `method` then?

Reply:
Ideally yes, but it's both a `class` method and an instance method, currently there's no way to do that with Object::Pad.